python/dgl/nn/pytorch/utils.py

```diff
@@ -395,3 +397,113 @@ def forward(self, feat_list):
             alpha = self.att(alpha).squeeze(-1)            # (N, num_layers)
             alpha = th.softmax(alpha, dim=-1)
             return (stacked_feat_list * alpha.unsqueeze(-1)).sum(dim=1)
+
+class LabelPropagation(nn.Module):
+    r"""Label Propagation from `Learning from Labeled and Unlabeled Data with Label
+    Propagation <http://mlg.eng.cam.ac.uk/zoubin/papers/CMU-CALD-02-107.pdf>`__
+
+    .. math::
+    \mathbf{Y}^{(t+1)} = \alpha \cdot \tilde{A} \mathbf{Y}^{(t)} + (1 - \alpha)
+    \mathbf{Y}^{(0)},
+    where unlabeled data is inferred by labeled data via propagation. :math:`\alpha`
```

Comment:
"where unlabeled data is inferred by labeled data via propagation" -> "where unlabeled data is initially set to zero and inferred from labeled data via propagation"